### PR TITLE
got basics of new review functionality

### PR DESCRIPTION
**Pull request by gusfel**
got basics of adding a new review.  i need to style it with some stars and the styling expectations of the business requirements.

### CODE REVIEW

File: Reviews/src/Components/NewReview/NewReview.jsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+  showModal() {
+    this.setState({
+      addPhotos: !this.state.addPhotos
+    });
+  }
+
+  submitReview() {
+    if (!this.checkState()) {
+      alert('please make sure all forms are filled out');
```
Comment (jincheng411) [inline Reviews/src/Components/NewReview/NewReview.jsx:79]: Would be ok if just let html to do the form validation?

File: Reviews/src/Components/NewReview/Rating.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  onChangeValue(event) {
+    let ratingObj = { rating: Number(event.target.value) };
+    this.props.updateState(ratingObj);
+  }
+
+  render() {
+    return (
+      <>
+        Overall Rating
+        <div onChange={this.onChangeValue}>
```
Comment (jincheng411) [inline Reviews/src/Components/NewReview/Rating.jsx:22]: there are some very nice rating code over codepen

File: Reviews/src/Components/ReviewApp.jsx [hunk 3]
```diff
@@ -14,10 +15,13 @@ class ReviewApp extends React.Component {
       ratings: {},
       loaded: false,
       displayedReviews: [],
+      newReview: false,
     };
 
     this.seeMoreReviews = this.seeMoreReviews.bind(this);
     this.getSort = this.getSort.bind(this);
+    this.showModal = this.showModal.bind(this);
```
Comment (jincheng411) [inline Reviews/src/Components/ReviewApp.jsx:23]: I think if you use arrow function directly inside the onClick, then you dont have to bind this

**Review by jincheng411 — APPROVED** [review]

File: Reviews/src/Components/ImageComponent.jsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import "/Users/gusfeliciano345/Desktop/Hack Reactor/bootcamp/FEC/soloFEC/thirdTimesACharm/Reviews/src/Components/NewReview/NewReview.css";
```
Comment (laenalea) [inline Reviews/src/Components/ImageComponent.jsx:2]: This is giving me a webpack bundling error in finding this file on my local system

File: Reviews/src/Components/NewReview/AddPhoto.jsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import css from './NewReview.css';
+
```
Comment (laenalea) [inline Reviews/src/Components/NewReview/AddPhoto.jsx:4]: I'm getting a style-loader error on this file, but I did run the npm install command in the Reviews folder before running webpack.

File: Reviews/src/Components/NewReview/NewReview.jsx [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+      name: '',
+      email: '',
+      photos: [],
+      characteristics: {},
+      addPhotos: false,
+    };
+
+    this.updateState = this.updateState.bind(this);
+    this.updateCharacteristics = this.updateCharacteristics.bind(this);
+    this.showModal = this.showModal.bind(this);
```
Comment (laenalea) [inline Reviews/src/Components/NewReview/NewReview.jsx:31]: Another webpack style-loader error originating from this line

**Comment by momo1113** [comment]
This is the errors I am getting.

```

> ERROR in ./src/Components/ImageComponent.jsx 24:0-138
> Module not found: Error: Can't resolve '/Users/gusfeliciano345/Desktop/Hack Reactor/bootcamp/FEC/soloFEC/thirdTimesACharm/Reviews/src/Components/NewReview/NewReview.css' in '/Users/momosun/workspace/hackreactor/FEC/thirdTimesACharm/Reviews/src/Components'
>  @ ./src/Components/NewReview/DisplayPhotos.jsx 3:0-51 8:44-58
>  @ ./src/Components/NewReview/NewReview.jsx 34:0-48 147:62-75
>  @ ./src/Components/ReviewApp.jsx 28:0-50 213:59-68
>  @ ./src/index.jsx 3:0-51 5:50-59
> 
> ERROR in ./src/Components/NewReview/AddPhoto.jsx 25:0-34
> Module not found: Error: Can't resolve 'style-loader' in '/Users/momosun/workspace/hackreactor/FEC/thirdTimesACharm/Reviews'
>  @ ./src/Components/NewReview/NewReview.jsx 31:0-38 143:63-71
>  @ ./src/Components/ReviewApp.jsx 28:0-50 213:59-68
>  @ ./src/index.jsx 3:0-51 5:50-59
> 
> ERROR in ./src/Components/NewReview/NewReview.jsx 25:0-34
> Module not found: Error: Can't resolve 'style-loader' in '/Users/momosun/workspace/hackreactor/FEC/thirdTimesACharm/Reviews'
>  @ ./src/Components/ReviewApp.jsx 28:0-50 213:59-68
>  @ ./src/index.jsx 3:0-51 5:50-59
> 
> 3 errors have detailed information that is not shown.
> Use 'stats.errorDetails: true' resp. '--stats-error-details' to show it.
> 
> webpack 5.24.2 compiled with 3 errors in 3982 ms
> 

```

**Review by laenalea — CHANGES_REQUESTED** [review]
I'm getting a few webpack compiling errors due being able to resolve style-loader in a few files, but it still compiled. Upon loading up the browser page for port 3000, I wasn't able to see the changes you made though. I'm only seeing a single horizontal bar across the top of the page.

File: Reviews/src/Components/NewReview/AddPhoto.jsx (hunk 5 above)
Comment (gusfel) [reply Reviews/src/Components/NewReview/AddPhoto.jsx:4]: try it again, i think i fixed the css link

File: Reviews/src/Components/NewReview/NewReview.jsx (hunk 6 above)
Comment (gusfel) [reply Reviews/src/Components/NewReview/NewReview.jsx:31]: try it again, it should be resolved now

**Review by momo1113 — APPROVED** [review]
All looks good. The modal didn't work the way I expected. When I filled out all the forms, submit button still give me alert `please make sure all forms are filled out`. That's the minor unexpected behavior needs to fix.

**Comment by gusfel** [comment]
> All looks good. The modal didn't work the way I expected. When I filled out all the forms, submit button still give me alert `please make sure all forms are filled out`. That's the minor unexpected behavior needs to fix.

try again, but make sure there is at least 50 characters in the body section

**Pull request merged**